TACC/Core-CMS
# FP-1417: Callout CSS to Style Child Elements via Tag not Class

### Overview

Refactor the "Callout" UI pattern to rely on markup instead of child classes.

<details><summary>Why?</summary>

The TACC Callout plugin will be deprecated ([FP-1415](https://jira.tacc.utexas.edu/browse/FP-1415)). The TACC Callout plugin allowed us to control the markup. Without it, markup is in control of a CMS editor. The CMS editor will create markup using plugins that pre-date the TACC Callout plugin. The markup from expected plugin combinations must support Callout pattern (ideally with minimal classes).

</details>

### Issues

[FP-1417](https://jira.tacc.utexas.edu/browse/FP-1417)

### Screenshots

#### Demos

These demos show styles for all expected markup and image combinations.

| Demo | Screenshots |
| - | - |
| [**Manual Pattern Library**][cep-ui] | ([demo screenshot](https://user-images.githubusercontent.com/62723358/147126769-38ce21fa-98f6-46b0-87b1-6a38f67c9e88.jpeg)) <br /> ([DjangoCMS structure](https://user-images.githubusercontent.com/62723358/147126767-a95bfabd-90f5-4d35-999e-a6fd8d1b868b.png)) |
| [**Automatic Pattern Library**][kss-ui] | ([demo screenshot](https://user-images.githubusercontent.com/62723358/147127339-d0d66fa4-ceea-4e73-91cd-7b754dd30ae3.jpeg)) |

#### Use Cases

These use cases have pre-cropped images. (To test non-cropped images, see "Demos".)

| [Frontera Use Case][page-ui]<br />(wide) | [Frontera Use Case][page-ui]<br />(narrow) | [BrainMap Use Case][page-ui]<br />(wide) | [BrainMap Use Case][page-ui]<br />(narrow) |
| - | - | - | - |
| ![FP-1417 Frontera System Wide](https://user-images.githubusercontent.com/62723358/147126890-6ee94aef-8abd-43dc-b0b9-8f411ddf0e7b.png) | ![FP-1417 Frontera System Narrow](https://user-images.githubusercontent.com/62723358/147126887-a7bb3d72-a933-48f9-a503-12299b634984.png) | ![FP-1417 BrainMap Sleuth Wide](https://user-images.githubusercontent.com/62723358/147126886-82d2e4a3-19cf-44bd-9234-d5cee7e3ebd0.png) | ![FP-1417 BrainMap Sleuth Narrow](https://user-images.githubusercontent.com/62723358/147126882-99ae92bb-0ba1-46a4-b2ad-a629db7abcf6.png) |

#### Link Interaction

A callout is not required to be a link, but when it is, hover and activation UI should exist.

| Default State | Hover State | Active State |
| - | - | - |
| ![FP-1417 State Default](https://user-images.githubusercontent.com/62723358/147127924-21706d9d-0d4c-4219-ba38-0feca357f179.png) | ![FP-1417 State Hover](https://user-images.githubusercontent.com/62723358/147127926-a30a83e9-0e3a-414f-8493-d4c4d9003c89.png) | ![FP-1417 State Active](https://user-images.githubusercontent.com/62723358/147127923-d337ea98-d459-4cb8-b7e8-11ca8e632294.png) |

</details>

### Testing

> Manual Pattern Demo[^1]: [cep.tacc.utexas.edu][cep-ui]
> Automated Pattern Demo[^2]: [tacc-wbomar-css-pattern-library-kss.surge.sh][kss-ui]

0. Use remote server or [Test CMS Changes](https://github.com/TACC/Core-CMS/wiki/Test-CMS-Changes) locally.
1. Review/Create plugin combo's with class `c-callout` on the outermost elements.
2. Check that UI—on wide screen—matches [wide Design UI (scroll down)][des-ui--wide] ([screenshot][des-ui-snap--wide])
3. Check that UI—on narrow screen—matches [narrow Design UI (scroll down)][des-ui--thin] ([screenshot][des-ui-snap--thin]).

[^1]: Uses a snippet to add `c-callout` CSS to live site.
[^2]: Testing ugly KSS pattern library: https://github.com/tacc-wbomar/Core-CMS-Pattern-Library-KSS.

[cep-ui]: https://cep.tacc.utexas.edu/design-system/ui-patterns/c-callout/
[kss-ui]: http://tacc-wbomar-css-pattern-library-kss.surge.sh/section-components.html#kss-fullscreen-kssref-components-callout
[des-ui--wide]: https://xd.adobe.com/view/f3dad8d1-88ee-40eb-b8da-667b00e4f876-a689/screen/ff0348ff-577d-450a-bef0-5326f3f7c705/specs/
[des-ui--thin]: https://xd.adobe.com/view/f3dad8d1-88ee-40eb-b8da-667b00e4f876-a689/screen/cce23001-6fdd-47fe-bf6b-9e8161f55c85/specs/
[des-ui-snap--wide]: https://user-images.githubusercontent.com/62723358/147128294-13c393a5-c6cc-4520-97bd-da7eeff13956.png
[des-ui-snap--thin]: https://user-images.githubusercontent.com/62723358/147128284-e120fea3-8b89-45ae-ad92-e00e280c8822.png
[page-ui]: https://dev.fronteraweb.tacc.utexas.edu/system/
